### ISSUE TITLE
Sync - fix 404 errors

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#711]
See https://github.com/nf-core/tools/issues/709#issuecomment-666683931

Some pipelines get a 404 error when we try to make the automatic PR:

```json
ERROR     Sync failed for nf-core/deepvariant:                       sync.py:385
GitHub API returned code 404:
{
    "message": "Not Found",
    "documentation_url": "https://developer.github.com/v3/pulls/#create-a-pull-request"
}
```

I don't know why this is happening, but we can try looping a few times with a sleep if we get specifically this error. Hopefully they will work on another go...?

**Comment by ewels** [comment]
Should compare these 404 error pipelines with the v1.9 release sync to see if there is any overlap

**Comment by ewels** [comment]
Issue for the v1.9 release sync is here: https://github.com/nf-core/tools/issues/548

Four out of the five repos that gave 404 errors this time also did for v1.9:

* deepvariant - **1.9 and 1.10**
* hlatyping - just 1.9
* mag - **1.9 and 1.10**
* mhcquant - **1.9 and 1.10**
* nascent - **1.9 and 1.10**

So this is **not random** as I had initially thought. Need to look into these pipelines to see what makes them different from the rest.

**Comment by ewels** [comment]
Ok, if you try to [make this PR manually for deepvariant](https://github.com/nf-core/deepvariant/compare/master...nf-core:TEMPLATE) you see that there is no shared history between the branches, so that might explain that one.

The others look fine though, so that doesn't explain everything.

**Comment by ewels** [comment]
Trying to list PRs between these two branches on these repos also gives a 404, eg: https://api.github.com/repos/nf-core/mhcquan/pulls?head=nf-core:TEMPLATE&base=dev

**Comment by ewels** [comment]
Ok wait, I think I might have found it:

Buried in the verbose log:

```
Found username and repo from remote: nf-core, mhcquan - git@github.com:nf-core/mhcquant.git
```

The code overwrites the git repo name from the checked out repo (bad) [here](https://github.com/ewels/nf-core-tools/blob/866a591fbf50f71166439874c8f014a81dcd0f97/nf_core/sync.py#L151-L169) - need to skip this if it's suppled (or just delete this chunk of code and make it required).

There must be something about the regex `github\.com[:\/]([^\/]+)/([^\/]+)$` and this particular remote (`git@github.com:nf-core/mhcquant.git`) - but I can't see what.

**Comment by ewels** [comment]
Ok, I don't know what was wrong with the regex but it doesn't matter - I have stripped that from the code as we didn't need it anyway and now everything seems to work 🎉

**Comment by ewels** [comment]
Fixed in #716
